### PR TITLE
Exclude `sourcecode` from scalameta parsers

### PR DESCRIPTION
**Pull request by nafg**
Because it pulls in `sourcecode` for Scala 2.13 even on Scala 3, this just wasted a lot of my time trying to figure out why I was getting bizarre errors caused by Scala 3 not being able to run Scala 2 macros.

### CODE REVIEW

**Comment by nafg** [comment]
Oh. Production code depends on other parts of scalameta. Perhaps I should add another exclusion then.

File: build.sbt [hunk 1]
```diff
@@ -325,7 +325,7 @@ lazy val zioHttpGen = (project in file("zio-http-gen"))
       `zio-test-sbt`,
       `zio-config`,
       scalafmt.cross(CrossVersion.for3Use2_13),
-      scalametaParsers
+      (scalametaParsers % Test)
         .cross(CrossVersion.for3Use2_13)
         .exclude("org.scala-lang.modules", "scala-collection-compat_2.13"),
```
Comment (nafg) [inline build.sbt]:
```suggestion
      scalametaParsers
        .cross(CrossVersion.for3Use2_13)
        .exclude("org.scala-lang.modules", "scala-collection-compat_2.13"),
        .exclude("com.lihaoyi", "sourcecode_2.13"),
```

File: build.sbt [hunk 2]
```diff
@@ -328,6 +328,7 @@ lazy val zioHttpGen = (project in file("zio-http-gen"))
       scalametaParsers
         .cross(CrossVersion.for3Use2_13)
         .exclude("org.scala-lang.modules", "scala-collection-compat_2.13"),
```
Comment (nafg) [inline build.sbt]:
```suggestion
        .exclude("org.scala-lang.modules", "scala-collection-compat_2.13")
```

**Comment by nafg** [comment]
I don't think the remaining failures have to do with this change

**Comment by nafg** [comment]
The commits should probably be squashed

**Review by 987Nabil — APPROVED** [review]

**Pull request merged**